avocado/utils/partition.py

```diff
@@ -242,22 +242,35 @@ def mount(self, mountpoint=None, fstype=None, args=''):
         # Update the fstype as the mount command passed
         self.fstype = fstype
 
+    def _get_pids_on_mountpoint(self, mnt):
+        """
+        Returns a list of processes using a given mountpoint
+        """
+        try:
+            cmd = "lsof " + mnt
+            out = process.system_output(cmd)
+            return [int(line.split()[1]) for line in out.splitlines()[1:]]
+        except OSError as details:
+            msg = 'Could not run lsof to identify processes using %s' % mnt
+            LOG.error(msg)
+            raise PartitionError(self, msg, details)
+        except process.CmdError as details:
+            msg = 'Failure from "%s" execution: %s' % cmd
```

Comment:
There is a problem here:
```
Traceback (most recent call last):
  File "/avocado/selftests/unit/test_utils_partition.py", line 110, in test_force_unmount_except
    with mock.patch('avocado.utils.partition.process.run') as run_mocked:
  File "/avocado/avocado/utils/partition.py", line 309, in unmount
    self._unmount_force(mountpoint)
  File "/avocado/avocado/utils/partition.py", line 269, in _unmount_force
    for pid in self._get_pids_on_mountpoint(mountpoint):
  File "/avocado/avocado/utils/partition.py", line 258, in _get_pids_on_mountpoint
    msg = 'Failure from "%s" execution: %s' % cmd
TypeError: not enough arguments for format string
```

I got this adding a specific test to pass through this code block.:
```python
    def test_force_unmount_except(self):
        with open("/proc/mounts") as proc_mounts_file:
            (...)
            with mock.patch('avocado.utils.partition.process.run') as run_mocked:
                run_mocked.side_effect = process.CmdError
                self.assertTrue(self.disk.unmount())

            self.assertEqual(proc.poll(), -9)   # Process should be killed -9
        (...)
```